compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4666,24 +4671,33 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr) {
             return;
         }
 
-        BUnionType unionType = (BUnionType) exprType;
         // Filter out the list of types which are not equivalent with the error type.
-        Map<Boolean, List<BType>> resultTypeMap = unionType.getMemberTypes().stream()
-                .collect(Collectors.groupingBy(memberType -> types.isAssignable(memberType, symTable.errorType)));
+        List<BType> errorAssignableTypes = new ArrayList<>();
+        List<BType> errorNotAssignableTypes = new ArrayList<>();
```

Comment:
:pick: I think we create arrays with a capacity of 10 by default. Some times arrays can be empty. Max size can be a count of union members. There is an opportunity to save memory here.

Reply:
The size of the array can be changed.